addons/web/static/src/views/kanban/kanban_renderer.js

```diff
@@ -443,11 +443,16 @@ export class KanbanRenderer extends Component {
      * @param {HTMLElement} [params.previous]
      */
     async sortGroupDrop(dataGroupId, { previous }) {
+        const originalGroups = [...this.props.list.groups];
         this.toggleProcessing(dataGroupId, true);
-
         const refId = previous ? previous.dataset.id : null;
-        await this.props.list.resequence(dataGroupId, refId);
-
+        try {
+            await this.props.list.resequence(dataGroupId, refId);
+        } catch (error) {
+            this.props.list.groups = originalGroups;
+            this.toggleProcessing(dataGroupId, false);
+            throw error;
+        }
         this.toggleProcessing(dataGroupId, false);
```

Comment:
```suggestion
        try {
            await this.props.list.resequence(dataGroupId, refId);
        } finally {
            this.toggleProcessing(dataGroupId, false);
        }   
```

Reply:
The internal state of the model should be recovered by the model itself, so I don't think the error should be caught here to reset the groups in their original order.  I would do that in _resequence instead.